### ISSUE TITLE
Overview map

### ISSUE THREAD

**Issue by GoogleCodeExporter** [alistairreilly/andors-trail#10]
```
It would be nice to have an in-game map of the world. Places that you've 
travelled to and their geographical relation to each other.

```

Original issue reported on code.google.com by `oskar.wi...@gmail.com` on 20 Nov 2010 at 10:08

**Comment by GoogleCodeExporter** [comment]
```
The map should show where dead ends of the map are and where its possible to 
proceed.
```

Original comment by `SamuelPl...@gmail.com` on 26 Nov 2010 at 10:43

**Comment by GoogleCodeExporter** [comment]
```
Some overworld travel could be possible. E.g. once someone had "cleaned" an 
area (by defeating each (type of) opponent at least once), overworld travel is 
enabled for that area (enabled for towns by default). 
Thereafter, travel could be done by a single click on the destination (if there 
is an end-to-end route through cleaned areas, including starting and ending 
position)
```

Original comment by `surrano` on 8 Dec 2010 at 10:58

**Comment by GoogleCodeExporter** [comment]
```
Yes I like the idea. Perhaps take a small amount of gold for saving the time it 
takes to move manually.

Remark:
There should be always an end to end route, as you would not see areas where 
you have not been yet. And all other areas should be connected.
```

Original comment by `SamuelPl...@gmail.com` on 8 Dec 2010 at 11:49

**Comment by GoogleCodeExporter** [comment]
```
Since this is a work in progress is a world map not possible as the world 
continues to expand, so how about we add a simple "Jump list" until the world 
is finished ?

For now, I think only towns should be on it. And it should be possible to jump 
to town from any location, even caves and dungeons.
Later, if needed, can we add travel distance restriction (eg 5 maps away) or a 
jump fee. Possible a combination with fee-free jumps within a map range and 
regular fee beyond that (maybe even make the fee variable according to range).
```

Original comment by `kims...@gmail.com` on 19 Dec 2010 at 5:59

**Comment by GoogleCodeExporter** [comment]
```
I think we should definitely include the locations between the towns in it. If 
the player looks unto the map he could figure out if he needs to go north or 
west if he wants to a specific town.

This is how currently the world looks like: 
  C..
  ..CC
  . CC
  .....
...  .

Where "C" are the Cities and "." the places between. I think it could be as 
easy as that with the following additional informations:
 - where am I? Highlight the place where I am.
 - can I go to each of the 4 directions or are there any restrictions?
 - is there an entry to a dungeon in this place?
 - where are dead ends of the map, that I should explore later?

Needs a dropdown which map to display if in a cave. Cave Map or World map.

Some places of the final map could look like:
|-----|-----|
|D    |     |
|  .-----C--|
|  |  |  |  |
|--|--|--|--|
|  |  |  |  |
|  .-----.  |
|  |  |    D|
|--|--|-----|
|?????|
|?????|
|?????|
|-----|

"D" - Dungeon entry   "C" - City   "?" - unknown Place   "-" connections

Just some thoughts.
```

Original comment by `SamuelPl...@gmail.com` on 19 Dec 2010 at 6:37

**Comment by GoogleCodeExporter** [comment]
```
How should we supply the graphics for the worldmap? Should we have some image 
that we distribute inside the game, or should we somehow build the worldmap 
dynamically from the combined images of all individual maps?

Would we prefer a simplified map (with only roads, cities and dungeons), for 
example, like this:
http://www.vickijoel.org/ldgames/blands_map2.jpg
or should we have a full graphical map that also includes forests and 
mountains? Like this:
http://www.yandros.com/RPG/Conan/conanmap-mongoose-1e.jpg

If we are looking at a graphical map that is not dynamically created, we will 
need someone to actually draw the map.
```

Original comment by `oskar.wi...@gmail.com` on 19 Dec 2010 at 9:21

**Comment by GoogleCodeExporter** [comment]
```
If someone has good map drawing skills it would be nice if he would draw them.

As we do not have too much place for details I would prefer an abstract map 
showing a tile for each map. With just the informations I stated above.
```

Original comment by `SamuelPl...@gmail.com` on 19 Dec 2010 at 9:40

**Comment by GoogleCodeExporter** [comment]
```
I was thinking of a drawn simplified world map with key landmarks. This could 
be extended to a few/more (dynamic?) sub-maps that are more detailed and more 
jump points.

Mockups of both ideas would properly be a good idea before a decision is made.
```

Original comment by `kims...@gmail.com` on 19 Dec 2010 at 10:52

**Comment by GoogleCodeExporter** [comment]
```
I don't think jump from dungeon depths is a good idea. Whoever runs out of 
healing material should definitely die a horrible death in the depths of a 
dungeon }:-)

Seriously, only peaceful travel should be allowed. As for travelling fee -- 
some combats while travelling are almost inevitable. And combats give xp + 
gold/items (more than sufficient for healing). Jumping would not give this xp + 
gold. You may consider it a fee already.
```

Original comment by `surrano` on 21 Dec 2010 at 8:01

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
I also think only peaceful traveling should be available. Perhaps there could 
be a coachmen in each city. I would think a fee of 50 to 100 gold and for very 
long travels for 200 to 400 gold would be good prices.
```

Original comment by `SamuelPl...@gmail.com` on 22 Dec 2010 at 10:32

**Comment by GoogleCodeExporter** [comment]
```
As far as the type of map to display, a zoomed out version of the map you see 
normally.  You could even do something similar on the server side of things, 
take a screen shot and use it as the map that the coachman would have to select 
where you want to go, if you want to go the coachman route. Either way you 
could use the screenshot and links on the landmarks.

As an example here is an old map of one of my games.
Note: Each tile is actually 32x32 pixels so as you can tell it is really zoomed 
out.
http://deltoria.techby2guys.com/images/k5map.gif
```

Original comment by `sdeva...@gmail.com` on 23 Dec 2010 at 2:27

**Comment by GoogleCodeExporter** [comment]
```
I see multiple things we can use from the map picture in #12:
 * Water, islands and bridges
 * Roads offer guidance between major locations, but not needed for all quest parts
 * Different ground texture (grass, beach, desert, rocky, mountain, rough, swamp)
 * Dynamic generation using game tiles

It also offer ideas:
 * Maze
 * Castle / stronghold
 * Remote areas only accessible by limit means, like boat or two-way teleport devices
 * Fun patterns

We could use few simple map tiles to show routes without any real textures to 
show areas explored as well as key/check points. (I got this idea from a flash 
game).

Example of such tiles:
1)4-way  2)blind  3)3-way
|--E--|  |-----|  |--E--|
|   P |  |   P |  |   P |
E     E  |     E  E     E
|     |  |     |  |     |
|--E--|  |-----|  |-----|
P = player, E = entry/exit. We could add "D = Dead" or "N = no route yet" (if 
it would conflict with the possible D = Dungeon).

The pictures below show a combination of these tiles, which are from the game.
1) Game shows the current map tile (E-points are blank) on every sub-area the 
player is on
2) Tap "Map" once and you get the complete area with tiles (Note: I have 
discovered all titles for this area. Undiscovered tiles where NOT shown)(Yellow 
dot is you)(Blue tiles are key/check points and entry/exit points for the area, 
hence the new area name shown near them. See next picture)
3) Tap "Map" again and you get the world view
```

Original comment by `kims...@gmail.com` on 23 Dec 2010 at 6:21

Attachments:
- [Arcuz interface.png](https://storage.googleapis.com/google-code-attachments/andors-trail/issue-10/comment-13/Arcuz interface.png)
- [Arcuz map1.png](https://storage.googleapis.com/google-code-attachments/andors-trail/issue-10/comment-13/Arcuz map1.png)
- [Arcuz map2.png](https://storage.googleapis.com/google-code-attachments/andors-trail/issue-10/comment-13/Arcuz map2.png)

**Comment by GoogleCodeExporter** [comment]
```
I like the Map 1 above, as an additional setting include NPC's related to a 
quest; additionally, selecting an NPC name in the quest log could either 
activate a compass on screen or simply a geographical placement on the map 
overview if the compass is too difficult.
(While in development identify in the map which areas are closed off or are 
otherwise inaccessible.)
```

Original comment by `jimmywsc...@gmail.com` on 1 Jan 2011 at 8:24

**Comment by GoogleCodeExporter** [comment]
```
i like the simple little maps and trails i like the design on that 1 above i 
real think there should be a map
```

Original comment by `breid3...@hotmail.com` on 11 Jan 2011 at 5:37

**Comment by GoogleCodeExporter** [comment]
```
Is every map screen uniform in size? If so, Map 1 in comment 13 would be 
perfect.
```

Original comment by `joeybea...@gmail.com` on 11 Jan 2011 at 3:48

**Comment by GoogleCodeExporter** [comment]
```
how about this type of map? of course a bit more detailed.....
```

Original comment by `michisch...@web.de` on 11 Jan 2011 at 4:18

Attachments:
- [Map.gif](https://storage.googleapis.com/google-code-attachments/andors-trail/issue-10/comment-17/Map.gif)

**Comment by GoogleCodeExporter** [comment]
```
I agree with wanting a map.  I especially like the idea of coachmen to 
transport between cities.
```

Original comment by `theGl...@bigfoot.com` on 16 Jan 2011 at 10:08

**Comment by GoogleCodeExporter** [comment]
```
I personally am partial to the more graphical map with a google maps style 
interface. Pinch to zoom, long-press to select an action for a location (ie: 
set waypoint, jump to city, placed stuff in trunk here, etch), navigation (a 
small translucent marker pointing towards the direction you should go on the 
tile to get to your desired waypoint/city), things like that.

Knowing that this is a major endeavor and not necessary at the moment a 
simplified map would be extremely nice in the short term.

Just my measly two cents..
```

Original comment by `divilsp...@gmail.com` on 17 Jan 2011 at 8:11

**Comment by GoogleCodeExporter** [comment]
```
I prefer mobile apps to be lightweight and, especially, light on power 
consumption. Zooming and stuff sounds a bit overreacting, but if it is not a 
real-time zoom, more like a tree-like structure, then I'm in.

E.g. a map in "Arcuz map2" style, and if you click on a polygon then a map 
similar to "Arcuz map1" or even "Map.gif" comes up. This can be iterated over 
multiple levels of course.

As for the mood of the game, I'd prefer artistic maps (Arcuz map2) to 
auto-generated tiled maps (other two).
```

Original comment by `surrano` on 18 Jan 2011 at 10:56

**Comment by GoogleCodeExporter** [comment]
```
my personal opinion:
"eye candy" maps are good for players who seldom play the game. hard core games 
are most times not interested in eye candy but in good and efficient gameplay.

therefore: yes, an artistic map with cool zoom in/out effects would really be 
sexy. but an autogenerated (=dull) map consumes much less space on the device 
and is (hopefully) muss faster to display. therefore i prefer the autogenerated 
map.

bzw: it seems that there is only one developer for the game. and i just can't 
wait to get more content (more maps, more quest, more story!). so please: do 
whatever you want - but do it that way which consumes less time :) (time to 
design more maps and quests).

i really believe that this game has great potential - actually i am playing it 
more often then "pocket legends" which is a commercial (and finished) 3d mmorpg.
```

Original comment by `erwr...@gmail.com` on 18 Jan 2011 at 6:34

**Comment by GoogleCodeExporter** [comment]
```
Speed and size should definitely be priorities. Looks are for gaming systems. 
Using phones as a platform goes hand in hand with being a minimalist. Although 
this games size is definitely fine. It hurts downloading a +20 mb game. I would 
say sound is another luxury that isn't necessary. 
```

Original comment by `Captain8...@gmail.com` on 18 Jan 2011 at 8:11

**Comment by GoogleCodeExporter** [comment]
```
how about each tile type is the color of one pixel in the overworld map. fast 
and easy.

coachmen between cities is good. I suggested 'Wyvern Wings" to return to lasty 
town, maybe a tekeport spell or item.
```

Original comment by `rqpa...@gmail.com` on 3 Feb 2011 at 6:10

**Comment by GoogleCodeExporter** [comment]
Original comment by `oskar.wi...@gmail.com` on 10 Feb 2011 at 6:43
- Added labels: **Priority-Medium**
- Removed labels: **Priority-Low**

**Comment by GoogleCodeExporter** [comment]
```
Zoom sounds cool, but I don't know if it will work in practice with the natural 
blockyness of the maps. It might be the best way, though, once the world map 
gets huge.

This isn't a bad way:
http://img354.imageshack.us/img354/3108/maprenewalnp2.jpg
Not the prettiest, but you can immediately recognize each "map" as a square on 
the big map. I don't know how it will look on a tiny phone. The Link's 
Awakening map idea is cool, however in that game each "square" was one 
screen-size; this game scrolls around within a map, so you won't be able to get 
much detail.
```

Original comment by `jch...@email.wm.edu` on 8 Apr 2011 at 5:14

**Comment by GoogleCodeExporter** [comment]
```
I'm thinking about this issue and wants to try to get something done.

I would like to create a big bitmap file from the existing tmx files and 
tileset.

Using the existing map reader seems to be enough to load all tmx and connect 
them.
But I have a problem identifying if a particular map is indoor or outdoor, and 
on which level it is (like in dungeons where you go deeper and deeper).

Is there any objection adding some level and location in the map properties, 
like the following patch for crossglen ?

diff --git a/res/xml/crossglen.tmx b/res/xml/crossglen.tmx
index 35c450b..44f2e1b 100644
--- a/res/xml/crossglen.tmx
+++ b/res/xml/crossglen.tmx
@@ -1,5 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <map version="1.0" orientation="orthogonal" width="30" height="30" tilewidth="3+ <properties>
+  <property name="level" value="0"/>
+  <property name="location" value="outdoor"/>
+ </properties>
  <tileset firstgid="1" name="tiles_1_1" tilewidth="32" tileheight="32">
   <image source="../drawable/map_tiles_1_1.png"/>
  </tileset>

```

Original comment by `oli.sa...@gmail.com` on 24 May 2011 at 6:16

**Comment by GoogleCodeExporter** [comment]
```
(part of the latest patch was hate by my copy-paste, so here it is)
diff --git a/res/xml/crossglen.tmx b/res/xml/crossglen.tmx
index 35c450b..44f2e1b 100644
--- a/res/xml/crossglen.tmx
+++ b/res/xml/crossglen.tmx
@@ -1,5 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <map version="1.0" orientation="orthogonal" width="30" height="30" tilewidth="32" tileheight="32">
+ <properties>
+  <property name="level" value="0"/>
+  <property name="location" value="outdoor"/>
+ </properties>
  <tileset firstgid="1" name="tiles_1_1" tilewidth="32" tileheight="32">
   <image source="../drawable/map_tiles_1_1.png"/>
  </tileset>

```

Original comment by `oli.sa...@gmail.com` on 24 May 2011 at 6:18

**Comment by GoogleCodeExporter** [comment]
```
In response to comment 27: Sure, that seems reasonable. I'm sure we can use 
that for other purposes also. I'm thinking skills that only work 
indoors/outdoors and so on.

Good idea. Let's add that for v0.6.10 or v0.6.7.
```

Original comment by `oskar.wi...@gmail.com` on 7 Jun 2011 at 6:44

**Comment by GoogleCodeExporter** [comment]
```
regarding transportation:

qasur stated here:
http://andors.techby2guys.com/viewtopic.php?f=4&t=859

Some way to move greater distances easier. I would suggest Vacor's "rift spell" 
be a one-way teleport to an inn you've already visited of your choice for like 
300 or 500 gold.

Or, a legendary necklace of teleport that does above, but only once every 3 or 
6 on-game hours.
```

Original comment by `SamuelPl...@gmail.com` on 9 Jul 2011 at 7:30

**Comment by GoogleCodeExporter** [comment]
```
Oskar I prefer the second of the two maps but for your game the first is best.  
I dont think your game world is big enough yet for the second type of map. Make 
it as big as morrowind lol. Seriously you need like another 5 towns at least. 
You need rivets and sea, and mountains. I also agree with coachmen idea
```

Original comment by `nezkeys1...@gmail.com` on 4 Sep 2011 at 3:15

**Comment by GoogleCodeExporter** [comment]
```
I have a question regarding the auto-generation of the world map. Isn't it - at 
least theoretically - possible that two "adjacent" maps actually overlap? I 
mean, like this:
----------
|        |
|     ------------
|     |          |
|  A  |          |
|     |     B    |
|  ----------    |
|  |        |    |
---|        |    |
   |   C    |----|
   |        |
   |        |
   ----------
This would make all generation attempts obsolete, right?
```

Original comment by `s.jano...@gmail.com` on 11 Sep 2011 at 6:06

**Comment by GoogleCodeExporter** [comment]
```
that is the reason the gapfiller maps were included.
```

Original comment by `SamuelPl...@gmail.com` on 11 Sep 2011 at 9:18

**Comment by GoogleCodeExporter** [comment]
```
In other words presently it's not an issue but it's not *impossible* by design. 
Only, let's say, discouraged?
```

Original comment by `s.jano...@gmail.com` on 14 Sep 2011 at 4:19

**Comment by GoogleCodeExporter** [comment]
```
In-game worldmap released in v0.6.12.
```

Original comment by `oskar.wi...@gmail.com` on 28 Oct 2012 at 7:35
- Changed state: **Fixed**
